storage/queue.go

```diff
@@ -400,8 +401,12 @@ func (bq *baseQueue) remove(index int) {
 // Exposed for testing only.
 func (bq *baseQueue) DrainQueue(clock *hlc.Clock) {
 	bq.Lock()
-	defer bq.Unlock()
-	for repl := bq.pop(); repl != nil; repl = bq.pop() {
+	repl := bq.pop()
```

Comment:
The idea here was to keep the queue locked so nothing else could be added.  We don't have the controls to stop a queue from queueing while at the same time processing all existing replicas.